crates/red_knot_python_semantic/src/types.rs

```diff
@@ -649,26 +651,35 @@ impl<'db> Type<'db> {
                         },
                     )
             }
-            (Type::ClassLiteral(..), Type::Instance(InstanceType { class }))
-                if class.is_known(db, KnownClass::Type) =>
-            {
-                true
-            }
             (Type::ClassLiteral(self_class), Type::SubclassOf(target_class)) => {
-                self_class.class.is_subclass_of(db, target_class.class)
+                self_class.class.is_subclass_of_base(db, target_class.base)
             }
-            (Type::SubclassOf(self_class), Type::SubclassOf(target_class)) => {
-                self_class.class.is_subclass_of(db, target_class.class)
+            (Type::Instance(self_class), Type::SubclassOf(target_class))
+                if self_class.class.is_known(db, KnownClass::Type) =>
+            {
+                self_class.class.is_subclass_of_base(db, target_class.base)
             }
             (
-                Type::SubclassOf(SubclassOfType { class: self_class }),
-                Type::Instance(InstanceType {
-                    class: target_class,
+                Type::SubclassOf(SubclassOfType {
+                    base: ClassBase::Class(self_class),
                 }),
+                Type::SubclassOf(SubclassOfType {
+                    base: ClassBase::Class(target_class),
+                }),
+            ) => self_class.is_subclass_of(db, target_class),
+            // C ⊆ type
+            // type[C] ⊆ type
+            // Though note that this works regardless of which metaclass C has, not just for type.
+            (
+                Type::ClassLiteral(ClassLiteralType { class: self_class })
+                | Type::SubclassOf(SubclassOfType {
+                    base: ClassBase::Class(self_class),
+                }),
+                Type::Instance(target_class),
```

Comment:
I think I'd either call the captured variable here `target_instance` or do something like this

```suggestion
                Type::Instance(InstanceType {class: target_class}),
```

xo that you can avoid the awkward `target_class.class` immediately below

Reply:
Done